### PR TITLE
Drag and Drop feature #67

### LINKED ISSUE #67: Drag and Drop feature
### Description

Adding some drag & drop feature to the kanban boards. Successfully completing and merging this feature will also complete the following: 
- #82 
- #74 

Additional cleanup items: 
- Ensuring that the task can only be edited when the slide-over is open 
- color themes
- new tasks appear in a container and are rendered correctly

Resources: 
https://www.robinwieruch.de/react-drag-and-drop/
https://rootstack.com/en/blog/how-do-i-use-drag-and-drop-react
https://codesandbox.io/s/react-beautiful-dnd-board-base-0dv9b?file=/src/dnd/reorder.js

### Proposed Solution (for features)

- Drag and drop solution from tutorial 
- Tasks are created and rendered within the associated column (based on user input and clicking action within the said column)
- CRUD Column Management
- CRUD Tasks
- Tasks are edited within Slideover 
- Comments Rendering within Slideover

### Tasks 
Note: this feature incorporates many of the functionalities and issues outstanding. Completing this issue will complete the others. 

-   [ ] Task 1
-   [ ] Task 2

### Related Issues or PRs (optional)

[If there are any related issues or pull requests, link them here]

### PR DESCRIPTION
**Pull request by AitchGrasso**
## Description

- Drag and drop functionality + 
- Kanban Design + 
- Column added to Data Services, Router, Server routes, & Controller + 
- TaskModal + 
- Overall site theming

## Related Issue

-   [ X ] Is this related to a specific issue? If so please specify: Closes #67 [Drag and Drop feature#67](https://github.com/snowballDevs/kandoo/issues/67)

## Type of Changes

Use a check for ✓ the following type of changes:

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|  ✓    | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

home
kanban

### After
Screenshot 2023-08-13 225716
Screenshot 2023-08-13 225923

## Testing Steps / QA Criteria

n/a

## Checklist:

-   [ X ] This PR is up to date with the development branch, and merge conflicts have been resolved
-   [ X ] My code follows the style guidelines of this project
-   [ ] I have executed npm run lint and resolved any outstanding errors. 
-   [ ] I have performed a self-review of my own code
-   [ X ] I have commented my code, particularly in hard-to-understand areas
-   [ X ] My changes generate no new warnings

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,6 +23,9 @@
     },
     "private": true,
     "dependencies": {
+        "@dnd-kit/utilities": "^3.2.1",
+        "@headlessui/react": "^1.7.16",
+        "@heroicons/react": "^2.0.18",
         "react-hook-form": "^7.45.1",
         "react-icons": "^4.10.1"
```
Comment (mjbramich) [inline package.json:30]: Can we get dependencies installed for the client, in the package.json within the client not on the root package.json.
And perhaps with only need one icon library?

**Review by mjbramich — CHANGES_REQUESTED** [review]

**Review by Michelle-932 — COMMENTED** [review]
Looks good to me. I agree with Dusty's comments.

File: package.json (hunk 1 above)
Comment (AitchGrasso) [reply package.json]: Removed from root package.json:  "@dnd-kit/utilities": "^3.2.1",
        "@headlessui/react": "^1.7.16",
        "@heroicons/react": "^2.0.18",
   Dnd-kit and headless moved to client package.json.

**Comment by AitchGrasso** [comment]
@mjbramich changes are ready for review.

**Comment by mjbramich** [comment]
Great Work! @AitchGrasso

**Review by mjbramich — APPROVED** [review]

**Pull request merged**